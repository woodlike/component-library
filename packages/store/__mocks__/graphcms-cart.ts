export const graphcmsCarts = [
  {
    title: 'Order Summary',
    shippingLabel: 'Shipping',
    totalLabel: 'Total',
    vatValueLabel: 'VAT Included',
    itemsLabel: 'articles',
  },
];
