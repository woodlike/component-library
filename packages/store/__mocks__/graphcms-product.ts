export const graphcmsProducts = [
  {
    remoteTypeName: 'Product',
    remoteId: 'ckg2cpheo08ey0150xpne15df',
    locale: 'en',
    stage: 'PUBLISHED',
    createdAt: '2020-10-09T14:34:36.491668+00:00',
    updatedAt: '2020-10-09T15:07:50.045468+00:00',
    publishedAt: '2020-10-11T12:13:27.546932+00:00',
    title: 'ALLI One Piece',
    modelTitle: 'One Piece',
    modelDescription:
      '* Seamless finish.\n' +
      '* Scoop neckline, flattering any bust cup.\n' +
      '* Internal bustier with removable soft cups.\n' +
      '* Double-lined for more support.\n' +
      '* Slight shape effect.',
    fitAndCoverageTitle: 'Fit & Coverage Guide',
    fitAndCoverageDescription:
      '* Moderate coverage.\n' +
      '* Lovely for all body shapes.\n' +
      '* Model is wearing the size small.\n' +
      '* If you fall between sizes, we suggest purchasing the size up as our fit is graded slightly smaller.\n' +
      '* Model’s measurements: Height 170 cm, Bust 86 cm, Waist 69 cm, Hips 94 cm, Dress size 34/36',
    fabricFeature: {
      remoteTypeName: 'FabricFeature',
      remoteId: 'ckg2c8yi0086i0108g6bpgk0d',
      locale: 'en',
    },
    productMarineProtection: {
      remoteTypeName: 'ProductMarineProtection',
      remoteId: 'ckg2c4yzc083d0108xqhpg0nj',
      locale: 'en',
    },
    id: 'Product:ckg2cpheo08ey0150xpne15df:en',
    parent: null,
    internal: {
      contentDigest: '6ac6dddaf5ca492f131f71be117a08df',
      type: 'GraphCMS_Product',
      counter: 29,
      owner: 'gatsby-source-graphcms',
    },
    children: [],
  },
  {
    remoteTypeName: 'Product',
    remoteId: 'ckg2e5f0809bs01086ngbgiun',
    locale: 'en',
    stage: 'PUBLISHED',
    createdAt: '2020-10-09T15:14:59.431347+00:00',
    updatedAt: '2020-10-09T15:14:59.431347+00:00',
    publishedAt: '2020-10-09T15:16:58.515117+00:00',
    title: 'ANAIS Top',
    modelTitle: 'Top',
    modelDescription:
      '* Reversible style.\n' +
      '* Seamless finish.\n' +
      '* Slightly lower scoop neckline.\n' +
      '* Bust seam for a perfect fit.\n' +
      '* Elastic underbust band for maximum support.',
    fitAndCoverageTitle: 'Fit & Coverage Guide',
    fitAndCoverageDescription:
      '* Moderate coverage.\n' +
      '* The top is flattering on both small and large busts.\n' +
      '* For a comfy feeling choose one size bigger.\n' +
      '* Model is wearing the size small.\n' +
      '* Model’s measurements: Height 170 cm, Bust 86 cm, Waist 69 cm, Hips 94 cm, Dress size 34/36\n',
    fabricFeature: {
      remoteTypeName: 'FabricFeature',
      remoteId: 'ckg2c8yi0086i0108g6bpgk0d',
      locale: 'en',
    },
    productMarineProtection: {
      remoteTypeName: 'ProductMarineProtection',
      remoteId: 'ckg2c4yzc083d0108xqhpg0nj',
      locale: 'en',
    },
    id: 'Product:ckg2e5f0809bs01086ngbgiun:en',
    parent: null,
    internal: {
      contentDigest: 'fe79f6b2ab8bff5e27591374f3f493a1',
      type: 'GraphCMS_Product',
      counter: 30,
      owner: 'gatsby-source-graphcms',
    },
    children: [],
  },
];

export const graphcmsFabric = [
  {
    remoteTypeName: 'FabricFeature',
    remoteId: 'ckg2c8yi0086i0108g6bpgk0d',
    locale: 'en',
    stage: 'PUBLISHED',
    createdAt: '2020-10-09T14:21:45.009544+00:00',
    updatedAt: '2020-10-09T15:37:38.977238+00:00',
    publishedAt: '2020-10-09T15:37:41.680711+00:00',
    internalTitle: 'Carvico Vita 390',
    compositionTitle: 'Composition',
    compositionDescription: '* 78% Nylon Econyl®\n* 22% Elastane',
    title: 'Fabric Features',
    fabricFeaturesDescription:
      '* Shape retention\n' +
      '* Excellent coverage (absolutely non-see-through)\n' +
      '* UPF 50+ UV protection factor\n' +
      '* Suntan creams & oils resistant\n' +
      '* Ultra chlorine resistant\n' +
      '* Soft and smooth\n' +
      '* Made in Italy',
    products: [
      {
        remoteTypeName: 'Product',
        remoteId: 'ckg2cpheo08ey0150xpne15df',
        locale: 'en',
      },
      {
        remoteTypeName: 'Product',
        remoteId: 'ckg2e5f0809bs01086ngbgiun',
        locale: 'en',
      },
    ],
    id: 'FabricFeature:ckg2c8yi0086i0108g6bpgk0d:en',
    parent: null,
    internal: {
      contentDigest: '3aa6356e4b188ac2e0680f5e509a605e',
      type: 'GraphCMS_FabricFeature',
      counter: 28,
      owner: 'gatsby-source-graphcms',
    },
    children: [],
  },
];

export const graphcmsMarineProtection = [
  {
    remoteTypeName: 'ProductMarineProtection',
    remoteId: 'ckg2c4yzc083d0108xqhpg0nj',
    locale: 'en',
    stage: 'PUBLISHED',
    createdAt: '2020-10-09T14:18:39.052694+00:00',
    updatedAt: '2020-10-09T14:18:39.052694+00:00',
    publishedAt: '2020-10-09T14:18:53.694103+00:00',
    title: 'Marine Protection',
    description:
      '* 1% for the Planet® donation.\n' +
      '* The donation goes to the environmental organization Healthy Seas®.\n' +
      '* Healthy Seas® removes ghost fishing nets from our oceans. \n' +
      '* The same fishing nets used to create part of the ECONYL® yarn.\n' +
      '* Sustainable - regenerative product.',
    products: [
      {
        remoteTypeName: 'Product',
        remoteId: 'ckg2cpheo08ey0150xpne15df',
        locale: 'en',
      },
      {
        remoteTypeName: 'Product',
        remoteId: 'ckg2e5f0809bs01086ngbgiun',
        locale: 'en',
      },
    ],
    id: 'ProductMarineProtection:ckg2c4yzc083d0108xqhpg0nj:en',
    parent: null,
    internal: {
      contentDigest: 'a0d68afc97a652ef6000ba2ab304521c',
      type: 'GraphCMS_ProductMarineProtection',
      counter: 27,
      owner: 'gatsby-source-graphcms',
    },
    children: [],
  },
];
