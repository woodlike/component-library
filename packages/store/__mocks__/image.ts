import { SourceSetProps } from '../src/gatsby';

export const srcsetMock = ([
  'https://cdn.shopify.com/s/files/1/0742/2733/products/25_500x500.jpg?v=1580468923',
  'https://cdn.shopify.com/s/files/1/0742/2733/products/25_800x800.jpg?v=1580468923',
  'https://cdn.shopify.com/s/files/1/0742/2733/products/25_1024x1024.jpg?v=1580468923',
] as unknown) as SourceSetProps[];
