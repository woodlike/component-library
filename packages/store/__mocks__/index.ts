export * from './graphcms-product';
export * from './header-data-mock';
export * from './image';
export * from './navigation-data-mock';
export * from './product-data-mock';
export * from './shopify-buy-cart';
export * from './shopify-collection';
export * from './shopify-page';
export * from './shopify-product';
