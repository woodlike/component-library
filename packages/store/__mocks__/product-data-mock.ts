export const productQueryMock = {
  allProductJson: {
    nodes: [
      {
        cartButton: 'Add to bag',
        ariaLabels: {
          select: 'size-variant-select',
        },
      },
    ],
  },
};
