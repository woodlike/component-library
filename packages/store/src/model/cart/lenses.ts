import { lensPath, set, view, Lens } from 'ramda';
import { ShopifyPriceV2, LineItem, Variant, Cart } from '.';
import { currency } from '..';

export const variantLens = lensPath(['variant']);
export const subtotalPriceV2Lens = lensPath(['amount']);
export const priceV2Lens = lensPath(['amount']);

export function formatCurrencyV2(lens: Lens, priceV2: ShopifyPriceV2) {
  const amount = view<ShopifyPriceV2, string>(lens, priceV2);
  return set(lens, currency(amount), priceV2);
}

export function reduceVariant(variant: Variant) {
  return {
    ...variant,
    priceV2: formatCurrencyV2(priceV2Lens, variant.priceV2),
  };
}

export function reduceLineItems(lineItems: LineItem[]): LineItem[] {
  return lineItems.map(lineItem => {
    const prevVariant = view<LineItem, Variant>(variantLens, lineItem);
    const variant = reduceVariant(prevVariant);
    return { ...lineItem, variant };
  });
}

export function reduceCart(cart: Cart) {
  const subtotalPriceV2 = formatCurrencyV2(
    subtotalPriceV2Lens,
    cart.subtotalPriceV2,
  );

  return {
    ...cart,
    subtotalPriceV2,
    lineItems: reduceLineItems(cart.lineItems),
  };
}
