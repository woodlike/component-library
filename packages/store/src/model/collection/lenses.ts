import { lensPath, set, view } from 'ramda';
import { ShopifyCollection } from '.';

export const collectionHandleLens = lensPath(['handle']);
export const collectionSlugLens = lensPath(['slug']);

export function setCollectionSlug(path: string, source: ShopifyCollection) {
  const handle = view<ShopifyCollection, string>(collectionHandleLens, source);
  return set(
    collectionSlugLens,
    `/${path}/${handle}`.replace(/\/\/+/g, '/'),
    source,
  );
}
