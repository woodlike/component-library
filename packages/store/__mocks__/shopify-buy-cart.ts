import { Cart } from '../src/model/cart';

export const cartMock: Cart = {
  __typename: 'Checkout',
  id: 'fooo=',
  ready: true,
  requiresShipping: true,
  note: null,
  paymentDue: '135.00',
  paymentDueV2: {
    amount: '135.0',
    currencyCode: 'EUR',
  },
  webUrl:
    'https://wdkl-ocean.myshopify.com/7422733/checkouts/3191b56c34ae73aca2cde2bdf5464daa?key=5664bc38bbe62a21f3e930ba6f9475ec',
  orderStatusUrl: null,
  taxExempt: false,
  taxesIncluded: true,
  currencyCode: 'EUR',
  totalTax: '0.00',
  totalTaxV2: { amount: '0.0', currencyCode: 'EUR' },
  lineItemsSubtotalPrice: { amount: '135.0', currencyCode: 'EUR' },
  subtotalPrice: '135.00',
  subtotalPriceV2: { amount: '135.0', currencyCode: 'EUR' },
  totalPrice: '135.00',
  totalPriceV2: { amount: '135.0', currencyCode: 'EUR' },
  completedAt: null,
  createdAt: '2020-11-15T15:25:35Z',
  updatedAt: '2020-11-20T16:52:59Z',
  email: null,
  discountApplications: [],
  appliedGiftCards: [],
  shippingAddress: null,
  shippingLine: null,
  customAttributes: [],
  order: null,
  lineItems: [
    {
      id: 'foo2==',
      title: 'ALLI One Piece - blue',
      variant: {
        id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8zMTM2NjgzNDE5MjQ3Mw==',
        title: 'S/M',
        price: '135.00',
        priceV2: { amount: '135.0', currencyCode: 'EUR' },
        presentmentPrices: [
          {
            price: { amount: '135.0', currencyCode: 'EUR' },
            compareAtPrice: { amount: '165.0', currencyCode: 'EUR' },
            hasNextPage: false,
            hasPreviousPage: false,
            variableValues: {
              id:
                'Z2lkOi8vc2hvcGlmeS9DaGVja291dC8zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYT9rZXk9NTY2NGJjMzhiYmU2MmEyMWYzZTkzMGJhNmY5NDc1ZWM=',
            },
          },
        ],
        weight: 195,
        available: true,
        sku: 'WDLK202013-2',
        compareAtPrice: '165.00',
        compareAtPriceV2: { amount: '165.0', currencyCode: 'EUR' },
        image: {
          id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvMTM2OTQzNDU5MzY5ODU=',
          src:
            'https://cdn.shopify.com/s/files/1/0742/2733/products/155.jpg?v=1580458979',
          altText: null,
        },
        selectedOptions: [
          {
            name: 'Size',
            value: 'S/M',
          },
        ],
        unitPrice: null,
        unitPriceMeasurement: {
          measuredType: null,
          quantityUnit: null,
          quantityValue: 0,
          referenceUnit: null,
          referenceValue: 0,
        },
        product: {
          id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzQ0MzQ1OTE1ODAyNDk=',
          handle: 'alli-one-piece-blue',
        },
      },
      quantity: 1,
      customAttributes: [
        {
          key: 'slug',
          value: '/products/alli-one-piece-blue',
        },
      ],
      discountAllocations: [],
    },
    {
      id:
        'Z2lkOi8vc2hvcGlmeS9DaGVja291dExpbmVJdGVtLzE2OTQxMDIzNjkwODQxMD9jaGVja291dD0zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYQ==',
      title: 'Sole One Piece Reversible - White Bird / Riad',
      variant: {
        id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8xNjk0MTAyMzY5MDg0MQ==',
        title: 'L',
        price: '135.00',
        priceV2: {
          amount: '135.0',
          currencyCode: 'EUR',
        },
        presentmentPrices: [
          {
            price: { amount: '135.0', currencyCode: 'EUR' },
            compareAtPrice: { amount: '165.0', currencyCode: 'EUR' },

            hasNextPage: false,
            hasPreviousPage: false,
            variableValues: {
              id:
                'Z2lkOi8vc2hvcGlmeS9DaGVja291dC8zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYT9rZXk9NTY2NGJjMzhiYmU2MmEyMWYzZTkzMGJhNmY5NDc1ZWM=',
            },
          },
        ],
        weight: 0,
        available: true,
        sku: 'WDLK3037',
        compareAtPrice: '165.00',
        compareAtPriceV2: { amount: '165.0', currencyCode: 'EUR' },
        image: {
          id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvNDgwNDQxNjI0MTc1Mw==',
          src:
            'https://cdn.shopify.com/s/files/1/0742/2733/products/one_suit_bandeau-weiss-front-2.jpg?v=1544040728',
          altText: null,
        },
        selectedOptions: [
          {
            name: 'Size',
            value: 'L',
          },
        ],
        unitPrice: null,
        unitPriceMeasurement: {
          measuredType: null,
          quantityUnit: null,
          quantityValue: 0,
          referenceUnit: null,
          referenceValue: 0,
        },
        product: {
          id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzE3MDUyMjk5NzU2NDE=',
          handle: 'sole-one-piece-reversible-white-bird',
        },
      },
      quantity: 1,
      customAttributes: [
        {
          key: 'slug',
          value: '/products/sole-one-piece-reversible-white-bird',
        },
      ],
      discountAllocations: [],
    },
    {
      id:
        'Z2lkOi8vc2hvcGlmeS9DaGVja291dExpbmVJdGVtLzMxMzY2ODM0MTkyNDczMD9jaGVja291dD0zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYQ==',
      title: 'ALLI One Piece - blue',
      variant: {
        id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8zMTM2NjgzNDE5MjQ3Mw==',
        title: 'S/M',
        price: '135.00',
        priceV2: { amount: '135.0', currencyCode: 'EUR' },
        presentmentPrices: [
          {
            price: { amount: '135.0', currencyCode: 'EUR' },
            compareAtPrice: { amount: '165.0', currencyCode: 'EUR' },
            hasNextPage: false,
            hasPreviousPage: false,
          },
        ],
        weight: 195,
        available: true,
        sku: 'WDLK202013-2',
        compareAtPrice: '165.00',
        compareAtPriceV2: { amount: '165.0', currencyCode: 'EUR' },
        image: {
          id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvMTM2OTQzNDU5MzY5ODU=',
          src:
            'https://cdn.shopify.com/s/files/1/0742/2733/products/155.jpg?v=1580458979',
          altText: null,
        },
        selectedOptions: [{ name: 'Size', value: 'S/M' }],
        unitPrice: null,
        unitPriceMeasurement: {
          measuredType: null,
          quantityUnit: null,
          quantityValue: 0,
          referenceUnit: null,
          referenceValue: 0,
        },
        product: {
          id: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzQ0MzQ1OTE1ODAyNDk=',
          handle: 'alli-one-piece-blue',
        },
      },
      quantity: 1,
      customAttributes: [
        { key: 'slug', value: '/products/alli-one-piece-blue' },
      ],
      discountAllocations: [],
    },
  ],
};
