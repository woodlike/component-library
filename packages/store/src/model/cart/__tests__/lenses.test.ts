import * as Lens from '../lenses';
import { cartMock } from '../../../../__mocks__';

describe('lenses', () => {
  describe('setCurrencyFormatV2', () => {
    it('should return a formatted pricev2 object with two decimals digits amount', () => {
      const priceV2 = cartMock.lineItems[0].variant.priceV2;
      expect(Lens.formatCurrencyV2(priceV2)).toStrictEqual({
        currencyCode: priceV2.currencyCode,
        amount: '135.00',
      });
    });

    it('should return a formatted amount with two decimal digits', () => {
      cartMock.lineItems.forEach(item => {
        const priceV2 = item.variant.priceV2;
        const { amount } = Lens.formatCurrencyV2(priceV2);
        const decimal = amount.split('.')[1];
        expect(decimal.length).toBe(2);
      });
    });
  });
});
