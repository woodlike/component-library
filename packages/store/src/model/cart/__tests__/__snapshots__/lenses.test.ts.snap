// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "__typename": "Checkout",
  "appliedGiftCards": Array [],
  "completedAt": null,
  "createdAt": "2020-11-15T15:25:35Z",
  "currencyCode": "EUR",
  "customAttributes": Array [],
  "discountApplications": Array [],
  "email": null,
  "id": "fooo=",
  "lineItems": Array [
    Object {
      "customAttributes": Array [
        Object {
          "key": "slug",
          "value": "/products/alli-one-piece-blue",
        },
      ],
      "discountAllocations": Array [],
      "id": "foo2==",
      "quantity": 1,
      "title": "ALLI One Piece - blue",
      "variant": Object {
        "available": true,
        "compareAtPrice": "165.00",
        "compareAtPriceV2": Object {
          "amount": "165.0",
          "currencyCode": "EUR",
        },
        "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8zMTM2NjgzNDE5MjQ3Mw==",
        "image": Object {
          "altText": null,
          "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvMTM2OTQzNDU5MzY5ODU=",
          "src": "https://cdn.shopify.com/s/files/1/0742/2733/products/155.jpg?v=1580458979",
        },
        "presentmentPrices": Array [
          Object {
            "compareAtPrice": Object {
              "amount": "165.0",
              "currencyCode": "EUR",
            },
            "hasNextPage": false,
            "hasPreviousPage": false,
            "price": Object {
              "amount": "135.0",
              "currencyCode": "EUR",
            },
            "variableValues": Object {
              "id": "Z2lkOi8vc2hvcGlmeS9DaGVja291dC8zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYT9rZXk9NTY2NGJjMzhiYmU2MmEyMWYzZTkzMGJhNmY5NDc1ZWM=",
            },
          },
        ],
        "price": "135.00",
        "priceV2": Object {
          "amount": "135.00",
          "currencyCode": "EUR",
        },
        "product": Object {
          "handle": "alli-one-piece-blue",
          "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzQ0MzQ1OTE1ODAyNDk=",
        },
        "selectedOptions": Array [
          Object {
            "name": "Size",
            "value": "S/M",
          },
        ],
        "sku": "WDLK202013-2",
        "title": "S/M",
        "unitPrice": null,
        "unitPriceMeasurement": Object {
          "measuredType": null,
          "quantityUnit": null,
          "quantityValue": 0,
          "referenceUnit": null,
          "referenceValue": 0,
        },
        "weight": 195,
      },
    },
    Object {
      "customAttributes": Array [
        Object {
          "key": "slug",
          "value": "/products/sole-one-piece-reversible-white-bird",
        },
      ],
      "discountAllocations": Array [],
      "id": "Z2lkOi8vc2hvcGlmeS9DaGVja291dExpbmVJdGVtLzE2OTQxMDIzNjkwODQxMD9jaGVja291dD0zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYQ==",
      "quantity": 1,
      "title": "Sole One Piece Reversible - White Bird / Riad",
      "variant": Object {
        "available": true,
        "compareAtPrice": "165.00",
        "compareAtPriceV2": Object {
          "amount": "165.0",
          "currencyCode": "EUR",
        },
        "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8xNjk0MTAyMzY5MDg0MQ==",
        "image": Object {
          "altText": null,
          "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvNDgwNDQxNjI0MTc1Mw==",
          "src": "https://cdn.shopify.com/s/files/1/0742/2733/products/one_suit_bandeau-weiss-front-2.jpg?v=1544040728",
        },
        "presentmentPrices": Array [
          Object {
            "compareAtPrice": Object {
              "amount": "165.0",
              "currencyCode": "EUR",
            },
            "hasNextPage": false,
            "hasPreviousPage": false,
            "price": Object {
              "amount": "135.0",
              "currencyCode": "EUR",
            },
            "variableValues": Object {
              "id": "Z2lkOi8vc2hvcGlmeS9DaGVja291dC8zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYT9rZXk9NTY2NGJjMzhiYmU2MmEyMWYzZTkzMGJhNmY5NDc1ZWM=",
            },
          },
        ],
        "price": "135.00",
        "priceV2": Object {
          "amount": "135.00",
          "currencyCode": "EUR",
        },
        "product": Object {
          "handle": "sole-one-piece-reversible-white-bird",
          "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzE3MDUyMjk5NzU2NDE=",
        },
        "selectedOptions": Array [
          Object {
            "name": "Size",
            "value": "L",
          },
        ],
        "sku": "WDLK3037",
        "title": "L",
        "unitPrice": null,
        "unitPriceMeasurement": Object {
          "measuredType": null,
          "quantityUnit": null,
          "quantityValue": 0,
          "referenceUnit": null,
          "referenceValue": 0,
        },
        "weight": 0,
      },
    },
    Object {
      "customAttributes": Array [
        Object {
          "key": "slug",
          "value": "/products/alli-one-piece-blue",
        },
      ],
      "discountAllocations": Array [],
      "id": "Z2lkOi8vc2hvcGlmeS9DaGVja291dExpbmVJdGVtLzMxMzY2ODM0MTkyNDczMD9jaGVja291dD0zMTkxYjU2YzM0YWU3M2FjYTJjZGUyYmRmNTQ2NGRhYQ==",
      "quantity": 1,
      "title": "ALLI One Piece - blue",
      "variant": Object {
        "available": true,
        "compareAtPrice": "165.00",
        "compareAtPriceV2": Object {
          "amount": "165.0",
          "currencyCode": "EUR",
        },
        "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8zMTM2NjgzNDE5MjQ3Mw==",
        "image": Object {
          "altText": null,
          "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvMTM2OTQzNDU5MzY5ODU=",
          "src": "https://cdn.shopify.com/s/files/1/0742/2733/products/155.jpg?v=1580458979",
        },
        "presentmentPrices": Array [
          Object {
            "compareAtPrice": Object {
              "amount": "165.0",
              "currencyCode": "EUR",
            },
            "hasNextPage": false,
            "hasPreviousPage": false,
            "price": Object {
              "amount": "135.0",
              "currencyCode": "EUR",
            },
          },
        ],
        "price": "135.00",
        "priceV2": Object {
          "amount": "135.00",
          "currencyCode": "EUR",
        },
        "product": Object {
          "handle": "alli-one-piece-blue",
          "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0LzQ0MzQ1OTE1ODAyNDk=",
        },
        "selectedOptions": Array [
          Object {
            "name": "Size",
            "value": "S/M",
          },
        ],
        "sku": "WDLK202013-2",
        "title": "S/M",
        "unitPrice": null,
        "unitPriceMeasurement": Object {
          "measuredType": null,
          "quantityUnit": null,
          "quantityValue": 0,
          "referenceUnit": null,
          "referenceValue": 0,
        },
        "weight": 195,
      },
    },
  ],
  "lineItemsSubtotalPrice": Object {
    "amount": "135.0",
    "currencyCode": "EUR",
  },
  "note": null,
  "order": null,
  "orderStatusUrl": null,
  "paymentDue": "135.00",
  "paymentDueV2": Object {
    "amount": "135.0",
    "currencyCode": "EUR",
  },
  "ready": true,
  "requiresShipping": true,
  "shippingAddress": null,
  "shippingLine": null,
  "subtotalPrice": "135.00",
  "subtotalPriceV2": Object {
    "amount": "135.00",
    "currencyCode": "EUR",
  },
  "taxExempt": false,
  "taxesIncluded": true,
  "totalPrice": "135.00",
  "totalPriceV2": Object {
    "amount": "135.0",
    "currencyCode": "EUR",
  },
  "totalTax": "0.00",
  "totalTaxV2": Object {
    "amount": "0.0",
    "currencyCode": "EUR",
  },
  "updatedAt": "2020-11-20T16:52:59Z",
  "webUrl": "https://wdkl-ocean.myshopify.com/7422733/checkouts/3191b56c34ae73aca2cde2bdf5464daa?key=5664bc38bbe62a21f3e930ba6f9475ec",
}
`;
